File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/dto/response/ProblemResponseDTO.kt
package io.leonfoliveira.judge.api.dto.response

import io.leonfoliveira.judge.core.domain.entity.Problem

data class ProblemResponseDTO(
    val id: Int,
    val title: String,
    val description: String,
)

fun Problem.toResponseDTO(): ProblemResponseDTO {
    return ProblemResponseDTO(
        id = this.id,
        title = this.title,
        description = this.description,
    )
}

File: applications/webapp/src/core/port/dto/response/problem/ProblemResponseDTO.ts
import { AttachmentResponseDTO } from "@/core/port/dto/response/attachment/AttachmentResponseDTO";

export type ProblemResponseDTO = {
  id: string;
  createdAt: string;
  updatedAt: string;
  letter: string;
  color: string;
  title: string;
  description: AttachmentResponseDTO;
  timeLimit: number;
  memoryLimit: number;
  version: number;
};
